### ISSUE TITLE
[Feature Request] Option to Hide shorts below a chosen length

### ISSUE THREAD

**Issue by ACR-Jeff** [yuliskov/SmartTube#940]
Option to Hide shorts below a chosen length. I have the Hide shorts from Subscriptions option checked/enabled, Yet this option is only hiding videos below 50 seconds, This is leaving shorts in subscriptions 51 seconds and up. It would be nice to have the option to hide anything below 1 minute. 
Thank you for such a great app and all your hard work put into it.

**Comment by CMDR-Sloma** [comment]
Fully agree, my subscription feed is filled with this YouTube cancer. Maybe more robust option allowing you to set the length or by hashtag since those are using #shorts tags.

**Comment by costinmx** [comment]
Agree, it's better to use hastag #shorts as a filter, than time counter, it would be more efective. There are some honest news videos under 1:00 minute mark. But hastagged "shorts" are portrait oriented videos for smartphone use only.
@CMDR-Sloma excellent description as a " Youtube cancer".

**Comment by CMDR-Sloma** [comment]
This would be perfect... However, I just checked something. I use a Firefox extension called "BlockTube" and I have different options there for blocking videos. 

I went and set it up to block everything with #shorts tag. What I found out is short, vertical videos are still showing up in the subscription feed, simply because some creators don't use or forgot to use the tag, or they put it in the description of the video instead.

Few examples:
https://www.youtube.com/watch?v=dxl4yQRsbkY  - no tag at all, anywhere.
https://www.youtube.com/watch?v=VMZ4CAy2afA - tags are in the title and description, but I can still see the video somehow (???)
https://www.youtube.com/watch?v=6EhAOzuJg78 - '#short' in the video title only, no hashtags.
https://www.youtube.com/watch?v=srQi8eii8Z0 - can't see shorts or short anywhere.

Some people just don't bother to use hashtags, and this is why filtering this way doesn't fully remove this crap. 
I think just having a time filter might work best, I was able to cut the #YouTubeCancer entirely, it could be implemented as an additional setting. I don't have anything against short videos in particular, but being forced to watch them on a wide / ultrawide screen just drives me nuts. 

Your app is really excellent, definitely going to use it anyway, but it would be perfect if it had a time filter option.

**Comment by costinmx** [comment]
I watch a lot of official channels and broadcasters, and a time filter would be totally ineficient for this kind of official media. There are a lof of short news videos, made for TV, under 1 minute mark, wich renders this filter inaplicable. 
Serious channels tag there own videos responsible. 
User private channels dont have to obey almost any convenience rules, so they dont mark as 'shorts'. Sometimes they even dont want that, so they cannot be filtered out. You will never find an optimum time to filter this out, cause they will always try to circumvent it. 'By time', you are not filtering any real 'shorts', but all videos under a certain lenght.
I strongly support a filter dual option, first by filtering #shorts (also from title), then by time.

**Comment by CMDR-Sloma** [comment]
> ... I strongly support a filter dual option, first by filtering #shorts (also from title), then by time.

Sorry, wasn't precise. That's what I had in mind, time filter as an extra setting, not as a prime tool. Both would help clear subs from this crap. I totally get that every person watches different stuff, that's why extra options can help. Hope this makes sense. Similar like one below:

![image](https://user-images.githubusercontent.com/46288478/148582379-d147d8ea-c5f0-41ed-9fb5-213609494fa2.png)

**Comment by costinmx** [comment]
I just got a new ideea, I hope it helps:  
Would it be possible to filter shorts by portrait mode of the video? 
Because I think it's the smartphone portrait orientating video aspect that it's most annoying on TV. 
If this would be possible, then filter could be more effective than lenght or expression filtering. 
Hope Yuri @yuliskov still chase this feature... :)

**Comment by popy2k14** [comment]
Please give an option to remove shorts from Subscription. Thx

**Comment by popy2k14** [comment]
That's a really good idea.
Me too! I hate to look at the time and jump over those crap shorts.

Please give us such an filter.
